tests/administration/users.test.js

```diff
@@ -3,54 +3,64 @@ const common = require('../common'),
+  	return browser.waitForExist('//tbody')
+      .then(() => browser.isExisting('//td[text()="readonly" and position()=1]/..//button[contains(@id,"cp")]'))
+      .then((exists) => !exists ? common.createNewUser('readonly', 'local', 'readonly', 'admin') : null )
+  	  .then(() => browser.waitForExist('//td[text()="readonly" and position()=1]/..//button[contains(@id,"cp")]', 5000))
+      .then(() => common.waitAndClick('//td[text()="readonly" and position()=1]/..//button[contains(@id,"cp")]'))
+  	  .then(() => browser.setValue('//td[text()="readonly" and position()=1]/..//input[@id="password"]', 'admin'))
+  	  .then(() => browser.click('//td[text()="readonly" and position()=1]/..//button[@type="submit"]'))
   	  .then(() => browser.waitForExist('//h1[@class="page-header"]')) 
       .then(() => browser.getText('//h1[@class="page-header"]'))
   	  .then((text) => expect(text).to.equal('System Users'));
   });
 
   it('should allow deletion of readonly user', () => {
-  	return browser.click('//td[contains(text(),"readonly") and position()=1]/..//button[contains(@id,"rm")]')
-  	  .then(() => browser.waitForExist('//td[contains(text(),"readonly") and position()=1]/..//button[contains(@id,"rm")]', 500, true))
-      .then(() => browser.isExisting('//td[contains(text(),"readonly") and position()=1]/..//button[contains(@id,"rm")]'))
-      .then((exist) => expect(exist).to.be.false);
+  	return common.waitAndClick('//td[text()="readonly" and position()=1]/..//button[contains(@id,"rm")]')
+  	  .then(() => browser.waitForExist('//td[text()="readonly" and position()=1]/..//button[contains(@id,"rm")]', 5000, true))
   });
 
   it('should allow creation of a new readonly user', () => {
   	return common.createNewUser('readonly','local','readonly','admin')
-      .then((exist) => expect(exist).to.be.true);
+      .then(() => browser.waitForExist('//td[text()="readonly" and position()=1]'));
   });
 
   it('should allow for creation and deletion of new admin user', () => {
     return common.createNewUser('test','local','admin','admin')
-      .then((exist) => expect(exist).to.be.true)
-      .then(() => browser.click('//td[contains(text(),"test") and position()=1]/..//button[contains(@id,"rm")]'))
-      .then(() => browser.waitForExist('//td[contains(text(),"test") and position()=1]/..//button[contains(@id,"rm")]', 500, true))
+      .then(() => browser.waitForExist('//td[text()="test" and position()=1]/..//button[contains(@id,"rm")]'))
+      .then(() => browser.waitUntil(
+        browser.getAttribute('//td[text()="test" and position()=1]/..//button[contains(@id,"rm")]', 'onclick')
+          .then((cl) => { return cl !== null; })
+        ,10000))
+      .then(() => browser.click('//td[text()="test" and position()=1]/..//button[contains(@id,"rm")]'))
+      .then(() => browser.waitForExist('//td[text()="test" and position()=1]/..//button[contains(@id,"rm")]', 5000, true))
   });
```

Comment:
As above, if we do not verify the newly created user, we just know that the form doesn't spit an error